community/kernel/src/main/java/org/neo4j/kernel/api/proc/ProcedureSignature.java

```diff
@@ -156,6 +156,9 @@ public int hashCode()
     private final List<FieldSignature> inputSignature;
     private final List<FieldSignature> outputSignature;
     private final Mode mode;
+    private final boolean deprecated;
+    private final String deprecatedBy;
+    public static final String NOT_DEPRECATED = "";
```

Comment:
Yes but shouldn't the default value live in `Procedure` then. What you want to check is that if `deprecatedBy` is set or not when you compile the procedure?